changelog.d/20241008_214456_30907815+rjmello_compute_client_function_methods.rst

```diff
@@ -0,0 +1,8 @@
+Added
+~~~~~
+
+- Added ``ComputeClient.register_function()`` and ``ComputeClient.delete_function()``
+  methods. (:pr:`NUMBER`)
+
+- Added ``ComputeFunctionDocument`` and ``ComputeFunctionMetadata`` data classes for
+  use with the ``ComputeClient.register_function()`` method. (:pr:`NUMBER`)
```

Comment:
I've wanted to enhance our doc build pipeline to give a rendered context for these. Because we're using `scriv` to avoid changelog conflicts, it's possible to... :bulb: :exclamation: 
We'll probably change style to use sphinx roles if I can make things work the way I think we can.